### PR TITLE
[43] Migrate guides to the main repo

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

https://github.com/GetStream/android-internal-board/issues/43

We decided to migrate only guides from https://github.com/GetStream/Android-Samples to the main repo. 

### 🛠 Implementation details

Merged the guides into one application to avoid code duplication and multiple launch targets.

### 🎨 UI Changes

https://user-images.githubusercontent.com/9600921/184360236-6d404fad-2d9e-4f10-aa5c-0d87cbe226a8.mp4

### 🧪 Testing

1. Run `stream-chat-android-ui-guides` app.
2. Check the guides

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -18,13 +18,10 @@ stream-chat-android-test                    @GetStream/android-developers-core
 stream-chat-android-ui-common               @GetStream/android-developers-ui
 stream-chat-android-ui-components           @GetStream/android-developers-ui
 stream-chat-android-ui-components-sample    @GetStream/android-developers-ui
```
Comment (bychkovdmitry) [inline .github/CODEOWNERS:20]: I've simplified this a bit as some of the modules use Compose and XML at the same time.

File: settings.gradle [hunk 2]
```diff
@@ -7,6 +7,7 @@ include (
 		':stream-chat-android-ui-common',
 		':stream-chat-android-ui-components',
 		':stream-chat-android-ui-components-sample',
+		':stream-chat-android-ui-guides',
```
Comment (bychkovdmitry) [inline settings.gradle:10]: Better naming suggestions are welcome. I understand that the "guides app" could also contain random samples, but we use the word "sample" a lot.

File: stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/catalog/compose/customattachments/ChannelsActivity.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import io.getstream.chat.android.compose.ui.attachments.StreamAttachmentFactories
+import io.getstream.chat.android.compose.ui.channels.ChannelsScreen
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.guides.catalog.compose.customattachments.factory.dateAttachmentFactory
+
+/**
+ * An Activity representing a self-contained channel list screen with custom attachment factories.
+ */
+class ChannelsActivity : AppCompatActivity() {
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/catalog/compose/customattachments/ChannelsActivity.kt:32]: The guides themselves are mostly copied as is from the "Android-Samples" repository. Only added some KDocs.

File: stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/catalog/uicomponents/customattachments/composer/ChannelsActivity.kt [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            quotedAttachmentFactories = listOf(
+                QuotedDateAttachmentFactory(),
+                DefaultQuotedAttachmentMessageFactory()
+            )
+        )
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        // Reset to the default values not to affect other samples
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/catalog/uicomponents/customattachments/composer/ChannelsActivity.kt:65]: Hope this cleanup won't distract the users.

File: stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/login/LoginActivity.kt [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        )
+    }
+
+    /**
+     * Initializes the SDK with the given user and navigates to the catalog screen.
+     *
+     * @param loginUser The user credentials to login with.
+     */
+    private fun login(loginUser: LoginUser) {
+        ChatClient.instance().connectUser(
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/login/LoginActivity.kt:178]: The idea is that guides and samples should share as much logic as possible, so SDK initilization and logic is done only once. So, if we need a new guide we don't need to care about that.

File: settings.gradle (hunk 2 above)
Comment (adasiewiczr) [reply settings.gradle:10]: Are we going to keep only ui guides here? Maybe `stream-chat-android-guides` is a better name?

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -18,13 +18,10 @@ stream-chat-android-test                    @GetStream/android-developers-core
...
-
-# Compose
-stream-chat-android-compose             @GetStream/android-developers-ui
-stream-chat-android-compose-sample      @GetStream/android-developers-ui
-
-# UI testing
-stream-chat-android-ui-uitests          @GetStream/android-developers-ui
+stream-chat-android-compose                 @GetStream/android-developers-ui
+stream-chat-android-compose-sample          @GetStream/android-developers-ui
+stream-chat-android-ui-guides               @GetStream/android-developers-ui
```
Comment (adasiewiczr) [inline .github/CODEOWNERS:23]: I would rather change it to `@GetStream/android-developers` as everyone should be responsible for guides

File: stream-chat-android-ui-guides/detekt-baseline.xml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<SmellBaseline>
+  <ManuallySuppressedIssues/>
+  <CurrentIssues>
+    <ID>ComplexMethod:LoginUsers.kt$LoginUsers$fun createUsers(): List&lt;LoginUser></ID>
```
Comment (adasiewiczr) [inline stream-chat-android-ui-guides/detekt-baseline.xml:5]: @JcMinarro should we add special rules for guides? It seems most of them are Compose related 🤔

File: stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/catalog/uicomponents/customattachments/composer/ChannelsActivity.kt (hunk 4 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-guides/src/main/java/io/getstream/chat/android/guides/catalog/uicomponents/customattachments/composer/ChannelsActivity.kt:65]: Maybe we can create a helper object, do the cleanup there and just call the function? We are doing those cleanups in multiple places

**Review by adasiewiczr — COMMENTED** [review]
Good job, added a few small comments

File: stream-chat-android-ui-guides/detekt-baseline.xml (hunk 7 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-guides/detekt-baseline.xml:5]: I've added the Compose part of the config

**Review by adasiewiczr — APPROVED** [review]
Great job! Works as expected

**Pull request merged**